src/widgets/range-slider/range-slider.js

```diff
@@ -162,47 +167,35 @@ rangeSlider({
  *   })
  * );
  */
-export default function rangeSlider(
-  {
-    container,
-    attributeName,
-    min,
-    max,
-    templates = defaultTemplates,
-    cssClasses: userCssClasses = {},
-    step = 1,
-    pips = true,
-    precision = 2,
-    tooltips = true,
-    autoHideContainer = true,
-  } = {}
-) {
-  if (!container) {
-    throw new Error(usage);
-  }
-
-  const containerNode = getContainerNode(container);
-  const cssClasses = {
-    root: cx(bem(null), userCssClasses.root),
-    header: cx(bem('header'), userCssClasses.header),
-    body: cx(bem('body'), userCssClasses.body),
-    footer: cx(bem('footer'), userCssClasses.footer),
+export default function rangeSlider(params = {}) {
+  const widgetParams = {
+    container: '',
+    step: 1,
+    pips: true,
+    tooltips: true,
+    autoHideContainer: true,
+    cssClasses: {},
+    renderState: {},
+    templates: defaultTemplates,
+    ...params,
   };
 
-  const specializedRenderer = renderer({
-    containerNode,
-    step,
-    pips,
-    tooltips,
-    renderState: {},
-    templates,
-    autoHideContainer,
-    cssClasses,
-  });
+  if (!widgetParams.container) {
+    throw new Error(usage);
+  }
 
   try {
-    const makeWidget = connectRangeSlider(specializedRenderer);
-    return makeWidget({ attributeName, min, max, precision });
+    const makeWidget = connectRangeSlider(renderer);
```

Comment:
The nice thing is that we can even move that out: having a single instance is possible.